Cas23 addendum

blaisedias opened the pull request:
Store the share protocol used publish a nexus, so that correct actions can be determined at unpublish.

Commit:
Record the share protocol used to publish the nexus.

This so that the nexus can use the appropriate action at unpublish.

chriswldenyer reviewed (approved):
I was thinking of doing exactly that.

Commit:
Check and handle nexus share protocol at unpublish.

When a nexus is unpublished set the nexus share protocol to None.
Prior to that check that the share protocol is valid.

chriswldenyer commented on mayastor/src/bdev/nexus/nexus_share.rs:
```diff
@@ -100,6 +100,15 @@ impl Nexus {
+                let sp = match self.share_protocol {
+                    ShareProtocol::Nbd => Some(ShareProtocol::Nvmf),
+                    ShareProtocol::Iscsi => Some(ShareProtocol::Nvmf),
+                    ShareProtocol::Nvmf => Some(ShareProtocol::Nvmf),
```
Maybe we have a single function rather than duplicating this with the code around line 48? So that adding/removing a protocol can be done in fewer places.

chriswldenyer replied:
Actually, I suppose that's not worth changing ATM. My changes will affect that part of the code anyway.

blaisedias replied:
https://github.com/tjoshum/Mayastor/pull/3/commits/ca0905f02e30620700f17fcc6f942a4429211d4d

chriswldenyer reviewed (approved):
LGTM

Commit:
Common up validation checking for ShareProtocol value.

chriswldenyer reviewed (approved):
Looks Ok.
